src/main/java/com/codingiscaring/VendingMachine.java

```diff
@@ -4,51 +4,39 @@
 
 public class VendingMachine {
 
-    private List<Item> items;
-    private double cash = 100.00;
+    private final Casher cash = new Casher(100);
+    private final Inventory inventory;
 
     public VendingMachine(List<Item> items) {
-        this.items = items;
+        this.inventory = new Inventory(items);
     }
 
     public String vend(String code, double money) {
-        Item item = findItemBy(code);
 
-        if (item == null) {
-            return "Invalid selection! : Money in vending machine = " + DoubleUtils.twoDecimalsFormat(money);
-        }
+        try {
 
-        if (item.isOutOfStock()) {
-            return item.getName() + ": Out of stock!";
-        }
+            Item item = inventory.sell(code, money);
+            double change = cash.registerSell(item, money);
```

Comment:
```suggestion
            Item item = inventory.findItem(code);
            cash.registerSell(item, money);
            double change = cash.getChange(money);
```
And handle the exception InvalidSelectionException and add the to the message `: Money in vending machine = " + DoubleUtils.twoDecimalsFormat(money)`